deps/rabbit/src/rabbit_fifo_client.erl

```diff
@@ -320,29 +328,45 @@ discard(ConsumerTag, [_|_] = MsgIds,
 %%
 %% @returns `{ok, State}' or `{error | timeout, term()}'
 -spec checkout(rabbit_types:ctag(),
-               NumUnsettled :: non_neg_integer(),
                CreditMode :: rabbit_fifo:credit_mode(),
                Meta :: rabbit_fifo:consumer_meta(),
-               state()) -> {ok, state()} | {error | timeout, term()}.
-checkout(ConsumerTag, NumUnsettled, CreditMode, Meta,
-         #state{consumer_deliveries = CDels0} = State0) ->
+               state()) ->
+    {ok, ConsumerInfos :: map(), state()} |
+    {error | timeout, term()}.
+checkout(ConsumerTag, CreditMode, #{} = Meta,
+         #state{consumers = CDels0} = State0)
+  when is_binary(ConsumerTag) andalso
+       is_tuple(CreditMode) ->
     Servers = sorted_servers(State0),
-    ConsumerId = {ConsumerTag, self()},
-    Cmd = rabbit_fifo:make_checkout(ConsumerId,
-                                    {auto, NumUnsettled, CreditMode},
-                                    Meta),
+    ConsumerId = consumer_id(ConsumerTag),
+    Spec = case rabbit_fifo:is_v4() of
+               true ->
+                   case CreditMode of
+                       {simple_prefetch, 0} ->
+                           {auto, {simple_prefetch,
+                                   ?UNLIMITED_PREFETCH_COUNT}};
+                       _ ->
+                           {auto, CreditMode}
+                   end;
+               false ->
+                   case CreditMode of
+                       {credited, _} ->
+                           {auto, 0, credited};
+                       {simple_prefetch, 0} ->
+                           {auto, ?UNLIMITED_PREFETCH_COUNT, simple_prefetch};
+                       {simple_prefetch, Num} ->
+                           {auto, Num, simple_prefetch}
```

Comment:
It seems always life `auto` is used. Can this tuple element then be omitted? In other words, where is life `once` used in the checkout spec?